### PR TITLE
Stub out docs pages for `kolena.dataset`

### PR DESCRIPTION
**Pull request by gordonhart**
- Add "Datasets" Developer Guide section, separate from "Workflows" section
  - Stub out constituent quickstart and core concept files
- Add `kolena.dataset` stub section to API Reference docs

Fixes KOL-4395

### CODE REVIEW

**Comment by gordonhart** [comment]
Closed by #380

File: mkdocs.yml [hunk 1]
```diff
@@ -83,7 +96,7 @@ nav:
         - metrics/iou.md
       - Large Language Models:
         - metrics/contradiction-score.md
-        - metrics/vectara-HEM-score.md
+        - HEM Score: metrics/vectara-HEM-score.md
```
Comment (y27choi) [inline mkdocs.yml:99]: Is this change intentional?

**Review by y27choi — APPROVED** [review]
